fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/MessageValidator.java

```diff
@@ -44,36 +50,36 @@ public MessageValidatorBuilder checkValidLaoId(
     }
 
     /**
-     * Helper method to check that time in a Data is valid. The time value provided is assumed to be
-     * in Unix epoch time (UTC). Checks that: 1) time is not negative 2) time is not in the future
+     * Helper method to check that times are reasonably recent.The time values provided are assumed
+     * to be in Unix epoch time (UTC).
      *
-     * @param time time to check
-     * @throws IllegalArgumentException if time is invalid
+     * @param times time values to be checked
+     * @throws IllegalArgumentException if times are too far in the past
      */
-    public MessageValidatorBuilder checkValidTime(long time) {
-      if (time < 0) {
-        throw new IllegalArgumentException("Time cannot be negative");
-      } else if (isInFuture(time)) {
-        throw new IllegalArgumentException("Time cannot be in the future");
+    public MessageValidatorBuilder validPastTimes(Long... times) {
+      for (long time : times) {
+        if (time < VALID_PAST_TIME) {
+          throw new IllegalArgumentException("Time cannot be too far in the past");
+        }
+        if (time > CURRENT_TIME) {
+          throw new IllegalArgumentException("Time cannot be in the future");
+        }
       }
       return this;
     }
 
     /**
-     * Helper method to check that times in a Data are valid. The time values provided are assumed
-     * to be in Unix epoch time (UTC). Checks that time is valid and that the first time comes
-     * before the second time.
+     * Helper method to check that times in a Data are ordered. The time values provided are assumed
+     * to be in Unix epoch time (UTC).
      *
-     * @param beforeTime before time
-     * @param afterTime after time
-     * @throws IllegalArgumentException if a time is invalid or if the first time comes after the
-     *     second
+     * @param times time values to be checked
```

Comment:
You should specify in the method documentation that the times should be ordered and in what order